src/sonic-yang-models/yang-models/sonic-queue.yang

```diff
@@ -0,0 +1,88 @@
+module sonic-queue {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-queue";
+
+    prefix squeue;
+
+    import sonic-extension {
+        prefix sonic-ext;
+    }
+
+    import sonic-port {
+        prefix prt;
+    }
+
+    import sonic-scheduler {
+        prefix sch;
+    }
+
+    import sonic-wred-profile {
+        prefix wrd;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+	"QUEUE yang Module for SONiC OS";
+
+    revision 2021-04-01 {
+        description
+            "Initial revision";
+    }
+
+    container sonic-queue {
+
+        container QUEUE {
+
+            description "QUEUE part of config_db.json";
+            
+            list QUEUE_LIST {
+
+                key "ifname qindex";
+
+    //          sonic-ext:key-pattern "QUEUE|{ifname}|{qindex}"; //special pattern used for extracting keys from redis-key and populate the yang instance
+                                                                 // Total list instances = number(key1) * number(key2) * number(key3)
+
+                leaf ifname {
+                    type union {
+                        type leafref {
+                            path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:name";
+                        }
+                        type string {
+                            pattern "CPU";
+                        }
+                    }
+                    description "Interface name.";
+                }
+                // qindex format is (X) | (X-Y). X is start and Y is end index. 
+                // X and Y value depends on platform, example for physical ports 0-7 and
+                // for CPU port 0-48. Example qindex is 3-4
+                leaf qindex {
+                  //  sonic-ext:custom-validation ValidateQindexPattern;
+                    type string;
+                    description "Queue index on the interface.";
+                }
+
+                leaf scheduler {
+                    type leafref {
+                        path "/sch:sonic-scheduler/sch:SCHEDULER/sch:SCHEDULER_LIST/sch:name"; //Reference to SCHEDULER table
```

Comment:
@AshokDaparthi how do you expect to handle key reference notation from [swss-schema.md](https://github.com/Azure/sonic-swss/blob/master/doc/swss-schema.md)?

It seems that current YANG model won't be able to handle existing `config_db.json`:
**Example:**
```json
"QUEUE": {
    "Ethernet4|0": {
        "scheduler": "[SCHEDULER|scheduler.0]"
    },
    "Ethernet4|1": {
        "scheduler": "[SCHEDULER|scheduler.0]"
    },
    "Ethernet4|2": {
        "scheduler": "[SCHEDULER|scheduler.0]"
    },
    "Ethernet4|3": {
        "wred_profile": "[WRED_PROFILE|AZURE_LOSSLESS]",
        "scheduler": "[SCHEDULER|scheduler.1]"
    }
}

"SCHEDULER": {
    "scheduler.0": {
        "type": "DWRR",
        "weight": "14"
    },
    "scheduler.1": {
        "type": "DWRR",
        "weight": "15"
    }
}

"WRED_PROFILE": {
    "AZURE_LOSSLESS": {
        "red_max_threshold": "2097152",
        "wred_green_enable": "true",
        "ecn": "ecn_all",
        "green_min_threshold": "1048576",
        "red_min_threshold": "1048576",
        "wred_yellow_enable": "true",
        "yellow_min_threshold": "1048576",
        "green_max_threshold": "2097152",
        "green_drop_probability": "5",
        "yellow_max_threshold": "2097152",
        "wred_red_enable": "true",
        "yellow_drop_probability": "5",
        "red_drop_probability": "5"
    }
}

Which means that for scheduler/wred we expect to match:
1. scheduler -> scheduler.0
2. wred -> AZURE_LOSSLESS

and not:

1. scheduler -> SCHEDULER|scheduler.0
2. wred -> WRED_PROFILE|AZURE_LOSSLESS
```

Reply:
@praveen-li can you please elaborate what should be the appropriate YANG model for SONiC ABNF key reference?

How do we expect to handle this one?
```json
"BUFFER_PORT_EGRESS_PROFILE_LIST": {
    "Ethernet8": {
        "profile_list": "[BUFFER_PROFILE|egress_lossless_profile],[BUFFER_PROFILE|egress_lossy_profile]"
    },
```
```bash
root@sonic:/home/admin# redis-cli -n 4 HGETALL 'BUFFER_PORT_EGRESS_PROFILE_LIST|Ethernet8'
1) "profile_list"
2) "[BUFFER_PROFILE|egress_lossless_profile],[BUFFER_PROFILE|egress_lossy_profile]"
```

Reply:
@nazariig, @praveen-li  - We can discuss this over review call,  i am not sure pyang will take care leafref validation even if we give ABNF format. Else we should convert to leafref to pattern. But we will miss in this case leafref check's in YANG. As of now i am not sure is there anyway to resent in YANG.

Reply:
@nazariig - Back end changes to remove table ABNF format (example "[BUFFER_PROFILE|" )from field value is merged in latest master.